src/authorize.ts

```diff
@@ -78,6 +80,13 @@ export const authorize = (options: AuthorizeOptions): SocketIOMiddleware => {
       )
     }
     socket.decodedToken = decodedToken
+    if (onAuthentication != null) {
+      try {
+        socket.user = await onAuthentication(decodedToken)
+      } catch (err) {
+        return next(err)
+      }
```

Comment:
Is this try...catch useful ? What will happen, if we don't catch this error ?

Reply:
Well, the reason I am using a try...catch in here is to allow the user to throw a custom error and handle it on their client. And the reason I am calling `next(err)` in the catch block is because that is how I am seeing errors getting handled elsewhere in the library.